src/Orleans.Core/Logging/ErrorCodes.cs

```diff
@@ -4,7 +4,7 @@ namespace Orleans
     /// <summary>
     /// The set of error types used by the Orleans runtime libraries for logging errors.
     /// </summary>
-    internal enum ErrorCode
```

Comment:
Don't think our error codes should be public.
Makes it more likely app developers will use them, which can create logging confusion.
Suggest well defined ranges of error codes for core, extensions, and services, and move those used by extension libraries into those libraries.
Since existing codes may not conform to the defined ranges, we shouldn't enforce the ranges, but rather make it an 'orleans best practices' problem, wherein communicating the practice is probably sufficient.
See OrleansServiceBusErrorCode for example.